test/base/test_startpoint.py

```diff
@@ -55,3 +56,74 @@ def test_ubounded_startpoints(spmethod):
     ]:
         with pytest.raises(ValueError):
             spmethod(n_starts=n_starts, lb=lb_, ub=ub_)
+
+
+@pytest.fixture(params=[True, False])
+def check_fval(request) -> bool:
+    return request.param
+
+
+@pytest.fixture(params=[True, False])
+def check_grad(request) -> bool:
+    return request.param
+
+
+def test_resampling(check_fval: bool, check_grad: bool):
```

Comment:
Intended for use in additional tests? Otherwise `pytest.mark.parameterize` would be more concise.